src/user/user.controller.ts

```diff
@@ -1,93 +1,74 @@
-import { Body, Controller, Delete, Get, Param, Patch, Post, Render, Request, UseGuards } from '@nestjs/common';
+import {
+    Body,
+    Controller,
+    Delete,
+    Get,
+    HttpCode,
+    Param,
+    Patch,
+    Render,
+    Request,
+    UseGuards
+} from '@nestjs/common';
 import { ApiTags } from '@nestjs/swagger';
 import { ObjectId } from 'mongoose';
 import { Roles } from '../auth/roles/roles.decorator';
 import { RolesGuard } from '../auth/roles/roles.guard';
 import { JwtAuthGuard } from '../auth/strategies/jwt/jwt-auth.guard';
+import { returnUser } from './dto/return-user.dto';
 import { UpdateUserDto } from './dto/update-user.dto';
-import { User } from './entities/user.entity';
 import { Role } from './enums/role.enum';
+import { VerifyJWTGuard } from './guards/mailVerify-jwt.guard';
 import { UserService } from './user.service';
 
 @ApiTags('user')
 @Controller('user')
 export class UserController {
-  constructor(private readonly userService: UserService) { }
+    constructor(private readonly userService: UserService) {}
 
-  @Get()
-  @UseGuards(JwtAuthGuard, RolesGuard)
-  @Roles(Role.Admin)
-  async findAll(): Promise<any> {
-    return await this.userService.findAll();
-  }
+    @Get()
+    @UseGuards(JwtAuthGuard, RolesGuard)
```

Comment:
Do we require to use RolesGuard to verify the Role decorator?

Mabye can put them together or even create global api for it like `@IsUser` or `@IsAdmin`

```
            User	Moderator	Admin
IsUser	    ✓	   ✓	       ✓
IsModerator	x	   ✓	       ✓
IsAdmin	    x	   x	        ✓
```

Reply:
we might be able to, but the current way of handling this seems to be best practice already as it is in the nestjs docs
https://docs.nestjs.com/security/authorization

Reply:
> we might be able to, but the current way of handling this seems to be best practice already as it is in the nestjs docs https://docs.nestjs.com/security/authorization

Yes, we noticed this as well. However I think this design is quite bad.

Reply:
Do you think we should leave it like this?

Reply:
I will add this to nice to have...currently this is not necessary, but if we have time we can add this